streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -230,10 +233,8 @@ public TopologyTestDriver(final Topology topology,
 
         final MockConsumer<byte[], byte[]> consumer = new MockConsumer<>(OffsetResetStrategy.EARLIEST);
         stateDirectory = new StateDirectory(streamsConfig, mockTime);
-        final StreamsMetrics streamsMetrics = new StreamsMetricsImpl(
-            new Metrics(),
-            "topology-test-driver-stream-metrics",
-            Collections.<String, String>emptyMap());
+        metrics = new Metrics();
+        final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(metrics, "topology-test-driver-virtual-thread");
```

Comment:
"virtual" just in case people go looking for the actual thread in the thread dump. I also thought about using `Thead.currentThread()`, but it wouldn't necessarily be the same thread when the tests run.